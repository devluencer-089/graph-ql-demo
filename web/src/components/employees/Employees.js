import React from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import { withStyles } from 'material-ui/styles';

import EmployeeCard from './EmployeeCard';

export const query = gql`
  {
    query
    goes
    here
  }
`;

const styles = theme => ({
  list: {
    [theme.breakpoints.up('md')]: {
      display: 'flex'
    },
    flexWrap: 'wrap'
  }
});

function Employees(props) {
  // return <Query query={}/>

  return <div />;
}

export default withStyles(styles)(Employees);
