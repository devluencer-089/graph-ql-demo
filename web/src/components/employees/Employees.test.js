import React from 'react';
import { shallow } from 'enzyme';
import { Employees, query } from './Employees';

describe('Employees Component', () => {
  let data = {
    allEmployees: [
      {
        id: '3',
        level: 'SENIOR_DEVELOPER',
        firstname: 'Si',
        lastname: 'Tran'
      },
      {
        id: '2',
        level: 'SENIOR_DEVELOPER',
        firstname: 'Michael',
        lastname: 'Sewell'
      },
      {
        id: '1',
        level: 'ARCHITECT',
        firstname: 'Michael',
        lastname: 'Omann'
      }
    ]
  };

  it('should match snapshot', () => {
    let eut = shallow(<Employees data={data} classes={{}} />);

    expect(eut).toMatchSnapshot();
  });

  it('should contain employee cards', () => {
    let eut = shallow(<Employees data={data} classes={{}} />);

    expect(eut.find('WithStyles(EmployeeCard)')).toHaveLength(3);
  });

  describe('GraphQL Query', () => {
    it('should match snapshot', () => {
      expect(query).toMatchSnapshot();
    });
  });
});
