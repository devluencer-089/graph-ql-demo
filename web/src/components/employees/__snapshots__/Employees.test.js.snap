// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employees Component GraphQL Query should match snapshot 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "EmployeeQuery",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "version",
            },
            "selectionSet": undefined,
          },
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "allEmployees",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "level",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "firstname",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "lastname",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "phoneNumber",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 136,
    "start": 0,
  },
}
`;

exports[`Employees Component should match snapshot 1`] = `
<div>
  <div
    key="3"
  >
    <WithStyles(EmployeeCard)
      employee={
        Object {
          "firstname": "Si",
          "id": "3",
          "lastname": "Tran",
          "level": "SENIOR_DEVELOPER",
        }
      }
    />
  </div>
  <div
    key="2"
  >
    <WithStyles(EmployeeCard)
      employee={
        Object {
          "firstname": "Michael",
          "id": "2",
          "lastname": "Sewell",
          "level": "SENIOR_DEVELOPER",
        }
      }
    />
  </div>
  <div
    key="1"
  >
    <WithStyles(EmployeeCard)
      employee={
        Object {
          "firstname": "Michael",
          "id": "1",
          "lastname": "Omann",
          "level": "ARCHITECT",
        }
      }
    />
  </div>
</div>
`;
